src/ibek/runtime_cmds/commands.py

```diff
@@ -180,3 +184,146 @@ def generate_autosave(subst_file: Path = typer.Argument(GLOBALS.RUNTIME_SUBSTITU
+):
+    """
+    Expose the stdio of a process on a socket at unix:///tmp/stdio.sock.
+
+    This allows a local process to connect to stdio of the running process.
+    Use Ctrl+C to disconnect from the socket.
+
+    The following command will connect to the socket and provide interactive
+    access to the process:
+        socat UNIX-CONNECT:/tmp/stdio.sock -,raw,echo=0
+    """
+    asyncio.run(_expose_stdio_async(command))
+
+
+async def _expose_stdio_async(command: str):
+    # Check if the socket is already in use
+    socket_path = Path("/tmp") / "stdio.sock"
+    if socket_path.exists():
+        sys.stderr.write(f"Socket {socket_path} already exists. Exiting.\n")
+        raise typer.Exit()
+
+    # Create the socket and bind it to the path
+    server_socket = socket.socket(socket.AF_UNIX, socket.SOCK_STREAM)
+    server_socket.bind(str(socket_path))
+    server_socket.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
+    server_socket.listen(1)
+    server_socket.setblocking(False)
+    sys.stdout.write(f"Socket created at {socket_path}.\n")
+
+    # Start the process and pass the current environment variables
+    process = await asyncio.create_subprocess_shell(
+        command,
+        stdin=asyncio.subprocess.PIPE,
+        stdout=asyncio.subprocess.PIPE,
+        stderr=asyncio.subprocess.PIPE,
+        env={**os.environ},
+    )
+    sys.stdout.write(f"Process started with PID {process.pid}\n")
+
+    def handle_linefeed(char: bytes) -> bytes:
+        """
+        Handle line feed characters in the string.
+
+        When using socat in raw mode, line feeds do not send a carriage
+        return to the terminal. This function converts line feed characters
+        to carriage return + line feed characters for proper display.
+        """
+        if char == b"\n":
+            return b"\r\n"
+        return char
+
+    async def forward_stdout_and_socket(process, conn, conn_holder):
+        """Forward process stdout to sys.stdout and the socket if connected."""
+        while True:
+            char = await process.stdout.read(1)  # Read one character at a time
+            if not char:
+                break
+            sys.stdout.write(char.decode())
+            sys.stdout.flush()
+            if conn_holder["conn"]:
+                await asyncio.get_event_loop().sock_sendall(
+                    conn_holder["conn"], handle_linefeed(char)
+                )
+
+    async def forward_stderr_and_socket(process, conn, conn_holder):
+        """Forward process stderr to sys.stderr and the socket if connected."""
+        while True:
+            char = await process.stderr.read(1)  # Read one character at a time
+            if not char:
+                break
+            sys.stderr.write(char.decode())
+            sys.stderr.flush()
+            if conn_holder["conn"]:
+                await asyncio.get_event_loop().sock_sendall(
+                    conn_holder["conn"], handle_linefeed(char)
+                )
```

Comment:
DRY? Should this be a single function taking `process.stderr` and `sys.stderr` as args?

Reply:
for sure

Reply:
done